### ISSUE TITLE
0.8.14 :: Select-AzureSubscription error "must specify a non-null subscription name"

### ISSUE THREAD

**Issue by pcgeek86** [Azure/azure-powershell#244]
The `Select-AzureSubscription` command throws an exception when you specify an **invalid** subscription name, saying that the subscription name must be non-null. The exception message should be corrected to suggest to the end user that the subscription name is invalid, rather than being null, because clearly, it is not.

![image](https://cloud.githubusercontent.com/assets/466713/6531332/6d937d44-c3ed-11e4-8b51-bc42b31756ce.png)

In this example, I incorrectly typed the name of the subscription, and was thrown the exception.

```
PS C:\Users\Trevor> Select-AzureSubscription -SubscriptionName 'Parts Unlimited Sponsorship Program';
Select-AzureSubscription : Must specify a non-null subscription name.
Parameter name: name
At line:1 char:1
+ Select-AzureSubscription -SubscriptionName 'Parts Unlimited Sponsorsh ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Select-AzureSubscription], ArgumentException
    + FullyQualifiedErrorId : Microsoft.WindowsAzure.Commands.Profile.SelectAzureSubscriptionCommand
```

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell
http://trevorsullivan.net
http://twitter.com/pcgeek86

**Comment by markcowl** [comment]
@pcgeek86 Thanks for the feedback.  Scheduling this for an upcoming sprint

**Comment by markcowl** [comment]
This will be fixed in the 0.8.16 version due out this week - you can check the fix in the release or dev branch.  Please reopen if you see a recurrence.

**Comment by pcgeek86** [comment]
@markcowl Thanks! Great work!

Cheers,
Trevor Sullivan
Microsoft MVP: PowerShell
